vlog4j-client/src/main/java/org/vlog4j/client/picocli/VLog4jClientMaterialize.java

```diff
@@ -0,0 +1,161 @@
+	private List<String> queryStrings = new ArrayList<>();
+
+// TODO SaveModel
+//	/* group to save results */
+//	@ArgGroup(exclusive = false)
+//	private SaveModel saveModel;
+
+	@ArgGroup(exclusive = false)
+	private SaveQueryResult saveQueryResult = new SaveQueryResult();
+
+	/* group to print results */
+	@ArgGroup(exclusive = false)
+	private PrintQueryResults printQueryResult = new PrintQueryResults();
+
+	@Override
+	public void run() {
+		ExamplesUtils.configureLogging();
+		
+		System.out.println(queryStrings);
+
+//		if (saveModel.check() & saveQueryResult.check() & printQueryResult.check()) {
+//			saveModel.prepare();
+		if (saveQueryResult.checkConfiguration() & printQueryResult.checkConfiguration()) {
+			saveQueryResult.prepare();
+		} else {
+			System.exit(1);
+		}
+
+		System.out.println("Configuration:");
+
+		/* Configure rules */
+		final KnowledgeBase kb = new KnowledgeBase();
+		for (String ruleFile : ruleFiles) {
+			try {
+				RuleParser.parseInto(kb, new FileInputStream(ruleFile));
+				System.out.println("  --rule-file: " + ruleFile);
+			} catch (FileNotFoundException e) {
+				System.out.println("File not found: " + ruleFile + ". " + e.getMessage());
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse rule file: " + ruleFile + ". " + e.getMessage());
+			}
+		}
+
+		/* Configure queries (We throws parsing query errors before materialization) */
+		List<PositiveLiteral> queries = new ArrayList<>();
+		for (String queryString : queryStrings) {
+			try {
+				queries.add(RuleParser.parsePositiveLiteral(queryString));
+				System.out.println("  --query: " + queries.get(queries.size()-1).toString());
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse query: " + queryString.toString() + ". " + e.getMessage());
+				System.exit(1);
+			}
+		}
+
+		/* Print configuration */
+		System.out.println("  --log-file: " + logFile);
+		System.out.println("  --log-level: " + logLevel.toString());
+		System.out.println("  --chase-algorithm: " + chaseAlgorithm.toString());
+		System.out.println("  --timeout: " + ((timeout > 0) ? Integer.toString(timeout) : "none"));
+		/* Print what to do with the result */
+//		saveModel.print();
+		saveQueryResult.print();
+		printQueryResult.print();
+
+		try (Reasoner reasoner = new VLogReasoner(kb)) {
+			// logFile
+			reasoner.setLogFile(logFile);
+			// logLevel
+			reasoner.setLogLevel(logLevel);
+			// chaseAlgorithm
+			reasoner.setAlgorithm(chaseAlgorithm);
+			// timeout
+			if (timeout > 0) {
+				reasoner.setReasoningTimeout(timeout);
+			}
+
+			try {
+				System.out.println("Executing the chase ...");
+				reasoner.reason();
+			} catch (IOException e) {
+				System.out.println("Something went wrong. Please check the file paths inside the rule files.");
+				e.printStackTrace();
+			}
+
+//			// TODO save the model
+//			if (saveModel.saveModel) {
+//				System.out.println("Saving model ...");
+//			}
+
+			if (queries.size() > 0) {
```

Comment:
if the IO exception actually does occur above, the code below will still be reached.
I don't think we want to answer queries if there was an IO exception, or do we?